File: BackEnd/ExamManagement/DPD.HR.Business.Dto/Dto/PromotionTransferDto.cs
﻿using DPD.HumanResources.Entities.Entities;

namespace DPD.HumanResources.Dtos.Dto;

public class PromotionTransferDto
{
    public Guid Id {get;set;}
    public DateTime? CreatedDate {get;set;}     
    public int? Status {get;set;}
    public string? Description {get;set;}
    public Guid IdUserRequest {get;set;}
    public Guid IdUnit {get;set;}
    public Guid IdEmployee {get;set;}
    public string? UnitName {get;set;}
    public Guid? IdPositionEmployeeCurrent {get;set;}
    public string? PositionNameCurrent {get;set;}
    public bool IsTransfer {get;set;}
    public bool IsPromotion {get;set;}
    public bool IsHeadCount {get;set;}
    public Guid? IdCategoryPosition {get;set;}
    public string? NameCategoryPosition {get;set;}
    
    public List<WorkflowInstances>? WorkflowInstances { get; set; }
    public int CountWorkFlowStep { get; set; }
    public WorkflowStep? CurrentWorkFlowStep { get; set; }
    
}

File: FrontEnd/src/models/promotionTransferDTO.ts
import { WorkflowInstancesDTO } from "@models/workflowInstancesDTO";
import { WorkflowStepDTO } from "@models/workflowStepDTO";

export interface PromotionTransferDTO {
  id: string;
  createdDate: Date;
  status: number | null;
  description: string | null;
  idUserRequest: string;
  idUnit: string;
  unitName: string | null;
  idEmployee: string;
  idPositionEmployeeCurrent: string | null;
  positionNameCurrent: string | null;
  isTransfer: boolean | null;
  isPromotion: boolean | null;
  idCategoryPosition: string | null;
  nameCategoryPosition: string | null;
  isHeadCount: boolean | null;
  workflowInstances: WorkflowInstancesDTO[];
  countWorkFlowStep: number;
  currentWorkFlowStep: WorkflowStepDTO;
}
